source/class/cv/io/mqtt/Client.js

```diff
@@ -0,0 +1,263 @@
+
+    /**
+     * URL to the provided data
+     * @param name
+     * @return {String}
+     */
+    getProviderUrl: function (name) { return null; },
+
+    /**
+     * Mapping function the convert the data from the backend to a format the CometVisu data provider consumer can process.
+     * @param name {String}
+     */
+    getProviderConvertFunction : function (name) { return null; },
+
+    /**
+     * Set a subset of addresses the client should request initially (e.g. the ones one the start page).
+     * This can be used to increase the init state loading speed by sending an initial request with a smaller
+     * subset of addresses to the backend and send the rest later.
+     * @param addresses {Array}
+     */
+    setInitialAddresses: function(addresses) {
+      console.log('subscribe Initial: ', addresses);
+    },
+
+    /**
+     * Subscribe to the addresses in the parameter. The second parameter
+     * (filter) is optional
+     *
+     * @param addresses {Array?} addresses to subscribe to
+     * @param filters {Array?} Filters
+     *
+     */
+    subscribe : function (addresses, filters) {
+      addresses.forEach( value => this.client.subscribe(value) );
+    },
+
+    /**
+     * This function starts the communication by a login and then runs the
+     * ongoing communication task
+     *
+     * @param loginOnly {Boolean} if true only login and backend configuration, no subscription
+     *                            to addresses (default: false)
+     * @param credentials {Map} map with "username" and "password" (optional)
+     * @param callback {Function} call this function when login is done
+     * @param context {Object} context for the callback (this)
+     *
+     */
+    login : function (loginOnly, credentials, callback, context) {
+      let self = this;
+
+      function onConnect(param) {
+        self.setConnected(true);
+        if (callback) {
+          callback.call(context);
+        }
+      }
+
+      function onFailure(param) {
+        self.setConnected(false);
+        let n = cv.core.notifications.Router.getInstance();
+        n.dispatchMessage('cv.client.connection',{
+          title: 'MQTT: ' + qx.locale.Manager.tr('Connection error'),
+          message: param.errorMessage + "<br/>\nCode: " + param.errorCode,
+          severity: 'urgent',
+          unique: true,
+          deletable: false,
+        }, 'popup');
+      }
+
+      let options = {
+        timeout: 10,
+        onSuccess: onConnect,
+        onFailure: onFailure
+      };
+
+      if( null !== credentials && 'username' in credentials && null !== credentials.username ) {
+        options.userName = credentials.username;
+      }
+      if( null !== credentials && 'password' in credentials && null !== credentials.password ) {
+        options.password = credentials.password;
+      }
+
+      try {
+        this.client = new Paho.MQTT.Client(this._backendUrl, 'CometVisu_' + Math.random().toString(16).substr(2, 8));
```

Comment:
Please define this variable at the beginning if the members section. It would also be better to use and underscore for this to indicate that this this an internal (protected) variable: e,g,

```
members: {
   /*
   * @var {Paho.MQTT.Client}
   */
  _client: null,
  
  ....
  
  this._client = ...